azure-kit-starter/src/main/java/com/vaadin/azure/starter/ui/ClusterSupport.java

```diff
@@ -0,0 +1,174 @@
+     *
+     * @param listener
+     *            the listener to register.
+     */
+    public void setSwitchVersionListener(SwitchVersionListener listener) {
+        this.switchVersionListener = listener;
+    }
+
+    @Override
+    public void serviceInit(ServiceInitEvent serviceInitEvent) {
+        appVersion = System.getenv(ENV_APP_VERSION);
+        if (appVersion == null) {
+            getLogger().error(
+                    "Missing environment variable 'APP_VERSION'. ClusterSupport service not initialized.");
+            return;
+        }
+        getLogger().info(
+                "ClusterSupport service initialized. Registering RequestHandler with Application Version: "
+                        + appVersion);
+
+        // Register a generic request handler for all the requests
+        serviceInitEvent.addRequestHandler(createRequestHandler());
+    }
+
+    private RequestHandler createRequestHandler() {
+        return (vaadinSession, vaadinRequest, vaadinResponse) -> {
+            // Set the thread local instance
+            CurrentInstance.set(ClusterSupport.class, this);
+
+            vaadinSession.access(handleVersions(vaadinSession, vaadinResponse));
+
+            // If the current and the new versions do not match notify the user
+            return false;
+        };
+    }
+
+    private Command handleVersions(VaadinSession vaadinSession,
+            VaadinResponse vaadinResponse) {
+        return () -> {
+            // Always set the version cookie
+            Cookie versionCookie = getCookieByName(CURRENT_VERSION_COOKIE);
+            if (versionCookie == null
+                    || !versionCookie.getValue().equals(appVersion)) {
+                versionCookie = new Cookie(CURRENT_VERSION_COOKIE, appVersion);
+                versionCookie.setHttpOnly(true);
+                vaadinResponse.addCookie(versionCookie);
+            }
+
+            // Always check for the new version cookie
+            Cookie currentCookie = getCookieByName(UPDATE_VERSION_COOKIE);
+            if (currentCookie != null && !currentCookie.getValue().isEmpty()
+                    && !versionCookie.getValue()
+                            .equals(currentCookie.getValue())) {
+                vaadinSession.getUIs().forEach(ui -> {
+                    if (ui.getChildren().anyMatch(
+                            child -> (child instanceof VersionNotificator))) {
+                        return;
+                    }
+                    VersionNotificator notificator = new VersionNotificator(
+                            appVersion, currentCookie.getValue());
+                    notificator
+                            .addSwitchVersionEvent(createSwitchVersionEvent());
+                    // Show notificator
+                    ui.add(notificator);
+                });
+            }
+        };
+    }
+
+    private ComponentEventListener<VersionNotificator.SwitchVersionEvent> createSwitchVersionEvent() {
```

Comment:
A would name the method `createSwitchVersionEventListener`. Otherwise it seems to create and event

Reply:
Renamed the method to `createSwitchVersionEventListener`.